### DISCUSSION TITLE
Track progress of function in another file

### DISCUSSION THREAD

**Discussion by rohitsathish** [zauberzeug/nicegui#2100] [Q&A] [answered]
### Question

Lets say my main UI page is in ```home.py```. I have a long running function in ```utils.py``` that I call from ```home.py```. However, I would like to track the progress of the function in ```utils.py```, say print out different parts of the function and track the progress of a loop, and show it in the UI. 

Is this possible in NiceGUI? Is there a minimal example that I can look at?

**Comment by rodja** [comment]
I assume your long running function is cpu intensive. Then you'll need some kind of inter-process-communication. There is currently no such example. But would be very nice to add one. Any ideas on what kind of long running function could be suitable for this?

**Comment by rodja** [comment, marked as answer]
Oh! I totally forgot about our [progress example](https://github.com/zauberzeug/nicegui/blob/main/examples/progress/main.py). It uses a queue to do the communication and should be exactly what you need.
